check_source.py

```diff
@@ -152,10 +154,20 @@ def check_source_submission(self, source_project, source_package, source_revisio
             if devel_project:
                 if ((source_project != devel_project or source_package != devel_package) and
                    not (source_project == target_project and source_package == target_package)):
-                    # Not from proper devel project/package and not self-submission.
-                    self.review_messages['declined'] = 'Expected submission from devel package %s/%s' % (
-                        devel_project, devel_package)
-                    return False
+                    # check if the devel project & package are using scmsync
+                    # => waive the devel project source submission requirement
+                    meta = ET.fromstringlist(show_package_meta(self.apiurl, devel_project, devel_package))
+                    scm_sync = meta.find('scmsync')
+                    if (
+                            not scm_sync or
+                            (
+                                scm_sync and
+                                not scm_sync.text.startswith(f"https://src.opensuse.org/pool/{source_package}")
```

Comment:
Using `startswith` here will return false positives for URLs like `https://src.opensuse.org/pool/{source_package}-subpkg` or `https://src.opensuse.org/pool/{source_package}/../../hacker/pwned.git`

Reply:
if the devel project has a hacker/pwned reference, we have larger issues

Reply:
Right, I see this only checks the scmsync tag of the devel pkg, not the checked request. That should probably be done as well somewhere?

Reply:
The SR will be generated by a bot, thus I don't really see the need for that. It will not really make anything more secure, only hinder us in changing things later on.